src/Numerics/ODESolvers/MultirateInfinitesimalStepMethod.jl

```diff
@@ -186,27 +277,39 @@ function dostep!(Q, mis::MultirateInfinitesimalStep, p, time)
             α[i, :],
             β[i, :],
             γ[i, :],
-            d[i],
             dt;
             ndrange = length(realview(Q)),
             dependencies = (event,),
         )
         wait(array_device(Q), event)
 
-        fastrhs!.a = time + c̃[i] * dt
-        fastrhs!.b = (c[i] - c̃[i]) / d[i]
-
-        τ = zero(FT)
-        dτ = d[i] * dt / mis.nsubsteps
-        updatetime!(fastsolver, τ)
-        updatedt!(fastsolver, dτ)
-        # TODO: we want to be able to write
-        #   solve!(Q, fastsolver, p; numberofsteps = mis.nsubsteps)  #(1c)
-        # especially if we want to use StormerVerlet, but need some way to pass in `offset`
-        updatedt!(fastsolver, dτ)
-        for k in 1:(mis.nsubsteps)
-            dostep!(Q, fastsolver, p, τ, FT(1), realview(offset))  #(1c)
-            τ += dτ
+        # When d[i] = 0, we do not perform fast substepping;
+        # instead we just update the slow tendency
+        if abs(d[i]) < 1.e-10
+            Q .+= dt .* offset
+        else
+            fastrhs!.a = time + c̃[i] * dt
+            fastrhs!.b = (c[i] - c̃[i]) / d[i]
+
+            τ = zero(FT)
+            nsubstepsLoc = ceil(Int, nsubsteps * d[i])
+            dτ = d[i] * dt / nsubstepsLoc
```

Comment:
So this looks like the culprit for the test failure: previously we computed it as
```julia
dτ = d[i] * dt / mis.nsubsteps
```
This change probably makes more sense though.